packages/flutter/lib/src/material/navigation_drawer_theme.dart

```diff
@@ -0,0 +1,245 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:ui' show lerpDouble;
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/rendering.dart';
+import 'package:flutter/widgets.dart';
+
+import 'material_state.dart';
+import 'navigation_drawer.dart';
+import 'theme.dart';
+
+// Examples can assume:
+// late BuildContext context;
+
+/// Defines default property values for descendant [NavigationDrawer]
+/// widgets.
+///
+/// Descendant widgets obtain the current [NavigationDrawerThemeData] object
+/// using `NavigationDrawerTheme.of(context)`. Instances of
+/// [NavigationDrawerThemeData] can be customized with
+/// [NavigationDrawerThemeData.copyWith].
+///
+/// Typically a [NavigationDrawerThemeData] is specified as part of the
+/// overall [Theme] with [ThemeData.navigationDrawerTheme]. Alternatively, a
+/// [NavigationDrawerTheme] inherited widget can be used to theme [NavigationDrawer]s
+/// in a subtree of widgets.
+///
+/// All [NavigationDrawerThemeData] properties are `null` by default.
+/// When null, the [NavigationDrawer] will provide its own defaults based on the
+/// overall [Theme]'s textTheme and colorScheme. See the individual
+/// [NavigationDrawer] properties for details.
+///
+/// See also:
+///
+///  * [ThemeData], which describes the overall theme information for the
+///    application.
+@immutable
+class NavigationDrawerThemeData with Diagnosticable {
+  /// Creates a theme that can be used for [ThemeData.navigationDrawerTheme] and
+  /// [NavigationDrawerTheme].
+  const NavigationDrawerThemeData({
+    this.tileHeight,
+    this.backgroundColor,
+    this.elevation,
+    this.surfaceTintColor,
```

Comment:
Again, we should probably add the `shadowColor` here.

Reply:
Done